geop/src/main/scala/com/azavea/usace/programanalysis/geop/LayerReader.scala

```diff
@@ -0,0 +1,55 @@
+package com.azavea.usace.programanalysis.geop
+
+import geotrellis.raster.Tile
+import geotrellis.spark.io._
+import geotrellis.spark.{SpatialKey, TileLayerMetadata, _}
+import geotrellis.spark.io.s3.{S3AttributeStore, S3ValueReader}
+import geotrellis.vector.{Extent, MultiPolygon}
+
+import org.apache.spark.SparkContext
+
+object LayerReader {
+  /**
+    * Given a list of layer ids, a multipolygon and a spark context, returns
+    * a list of [[TileLayerRDD]]s cropped to the multipolygon.
+    *
+    * @param    rasterLayerIds  List of [[LayerId]]s to crop
+    * @param    multiPolygon    [[MultiPolygon]] to crop to
+    * @param    sc              [[SparkContext]] for the [[S3LayerReader]]
+    * @return                   List of [[TileLayerRDD]]s which includes only
+    *                           the tiles that either completely or partially
+    *                           overlap given multiPolygon.
+    */
+  def apply(
+    zoom: Int,
+    x: Int,
+    y: Int,
+    sc: SparkContext
+  ): Tile = {
+    val sKey: SpatialKey = SpatialKey(x, y)
+    val rdr = catalog(sc, zoom)
+
+    rdr.read(sKey)
+  }
+
+  /**
+    * Given a spark context, returns the correct catalog. This configures the
+    * next method with defaults.
+    *
+    */
+  def catalog(sc: SparkContext, zoom: Int): Reader[SpatialKey, Tile] =
+    catalog("azavea-datahub", "catalog", zoom)(sc)
+
+  def catalog(
+    bucket: String,
+    rootPath: String,
+    zoom: Int
+  )(implicit sc: SparkContext): Reader[SpatialKey, Tile] = {
+    val attributeStore = new S3AttributeStore(bucket, rootPath)
+    val vr = new S3ValueReader(attributeStore)
+    val layer = LayerId("nlcd-zoomed", zoom)
+    val reader: Reader[SpatialKey, Tile] = vr.reader(layer)
```

Comment:
Don't need to specify the data type on the left hand side since it will be computed from the right hand side.